filebeat/input/filestream/config.go

```diff
@@ -142,3 +144,57 @@ func (c *config) Validate() error {
 
 	return nil
 }
+
+// ValidateInputIDs checks all filestream inputs to ensure all have an ID and
+// the ID is unique. If there is any empty or duplicated ID, it logs an error
+// containing the offending input configurations and returns an error containing
+// the duplicated IDs.
+func ValidateInputIDs(inputs []*conf.C, logger *logp.Logger) error {
+	ids := make(map[string][]*conf.C)
```

Comment:
[Suggestion]
Rename this map to something that better describe its contents. It is a map of inputs (or configs) grouped by ID, so `ids` does not seem to be the better description to it.